### PR TITLE
feat: Check if an ENV var exists

### PR DESCRIPTION
**Pull request by ge0r**

### CODE REVIEW

File: internal/utils/system.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+func EnVar(key string) (string, error) {
+	if key == "" {
+		return "", exceptions.Wrap(exceptions.ErrArg, "empty argument")
+	}
+
+	val, ok := os.LookupEnv(key)
+
+	if val == "" {
+		if !ok {
+			return "", exceptions.Wrap(exceptions.ErrEnvVar, "variable not found")
```
Comment (zoispag) [inline internal/utils/system.go]:
```suggestion
			return "", exceptions.Wrap(exceptions.ErrEnvVar, "environment variable not found")
```

File: internal/utils/system.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+		}
+
+		return "", exceptions.Wrap(exceptions.ErrEnvVar, "variable is empty")
```
Comment (zoispag) [inline internal/utils/system.go]:
```suggestion
		return "", exceptions.Wrap(exceptions.ErrEnvVar, "environment variable is empty")
```

File: internal/utils/system.go (hunk 1 above)
Comment (zoispag) [inline internal/utils/system.go]: Why not check the `!ok` first? It is a distinct case and would keep the indent one level to the left.

**Review by zoispag — COMMENTED** [review]
Great job.
Please see some minor comments inline.

As far as the errors are related; do we need a whole `exceptions` package with a Wrap method and exported variables? For the zip util (PR #159) I used a different approach that I'd like to discuss with the rest of you.

File: internal/utils/system.go (hunk 1 above)
Comment (ge0r) [reply internal/utils/system.go]: Fair point! This structure is a leftover from a previous version of error throwing, it makes more sense to have it simpler now.

**Comment by drpaneas** [comment]
@ggkioulis I pushed a commit which simplifies your tests code and also I added checking for specific error messages. We will need to write tests for the errors package though. You can run `make cover` to see the current coverage.

File: internal/utils/system_test.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+			"",
+			"",
+			true,
+			exceptions.ErrArg.Error(),
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		os.Setenv(tt.key, tt.want)
```
Comment (ge0r) [inline internal/utils/system_test.go]: This does not allow the test to cover the real MISSING_KEY test case, since while testing we never get in this `GetEnv's` block:
```
if !ok {
		return "", exceptions.Wrap(exceptions.ErrEnvVar, "environment variable not found")
	}
```

File: internal/utils/system_test.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := GetEnv(tt.key)
+
+			// Test if error is expected to occur.
+			if tt.throwsError && err == nil {
```
Comment (ge0r) [inline internal/utils/system_test.go]: The way this expression is written, it never detects the case where `GetEnv` is wrongfully throwing an error when it should throw `nil`.

File: internal/exceptions/compare.go [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+// compareErrorMessages compares an error message with a string
+func compareErrorMessages(err error, wantSpecificError string) (bool, string) {
+	if strings.Contains(err.Error(), wantSpecificError) {
+		return true, wantSpecificError
+	}
+	return false, wantSpecificError
+}
+
+// Compare it's useful for testing specific error messages
+func Compare(err error, wantSpecificError string) (match bool, message string) {
```
Comment (ge0r) [inline internal/exceptions/compare.go:40]: This is neat!

File: internal/utils/system_test.go (hunk 3 above)
Comment (drpaneas) [reply internal/utils/system_test.go]: Trulio! Great catch !!

File: internal/utils/system_test.go (hunk 4 above)
Comment (drpaneas) [reply internal/utils/system_test.go:70]: There are 4 test cases in your table test data. Which one of those four is not covered? 

This expression checks that when you expect an error to happen (tc 2,3,4) then there is indeed an error returned, meaning it's not nil.

Comment (drpaneas) [reply internal/utils/system_test.go]: There are 4 test cases in your table test data. Which one of those four is not covered? 

This expression checks that when you expect an error to happen (tc 2,3,4) then there is indeed an error returned.

**Comment by drpaneas** [comment]
> Great job.
> Please see some minor comments inline.
> 
> As far as the errors are related; do we need a whole `exceptions` package with a Wrap method and exported variables? For the zip util (PR #159) I used a different approach that I'd like to discuss with the rest of you.

Your way of handling the errors is better approach (programmatically). Eventually we will do that at the second iteration for `romie`. However for the minimum valuable product this is not a priority, thus I would prefer to have this error wrapper to speed things up, so all the other golang newbies don't have to create their type of errors. 

tldr; go ahead with your error implementation, no need to downgrade your logic to the wrapper thing. We can of course discuss it further on the weekly meeting -- I just didn't want to keep you blocked until then.

File: internal/utils/system_test.go (hunk 4 above)
Comment (ge0r) [reply internal/utils/system_test.go]: Well, if in the future I change the implementation of `GetEnv` and make it wrongfully throw an error, when it does successfully get the value from an environment variable (and it should throw `nil` error), it will never be caught by the unit tests.

In order to cover this as well, the check would need to be
```if tt.throwsError != (err != nil)```

Comment (drpaneas) [reply internal/utils/system_test.go]: >  it will never be caught by the unit tests

You are wrong, unit tests would fail like this:

```shell
$ go test  -run GetEnv 
--- FAIL: ExampleGetEnv (0.00s)
got:
I am making an error on purose
want:
some-value
FAIL
exit status 1
FAIL    github.com/romie-gr/romie/internal/utils        0.306s
```

They fail due to `ExampleGetEnv()` ;) 

Hahaha, who thought an example would save the day? :D 

Anyway, we cannot rely on pure luck of example implementations. So, in the end you are right; this scenario is missing, but your way of handling this is very nasty, e.g.`if tt.throwsError != (err != nil)`. Why? Because my brain is too stupid to understand such complexity.

I think now I understand how your function works. I rewrote the tests to make the logic easier to understand. Let me know your thoughts.
